src/connector/src/parser/debezium/simd_json_parser.rs

```diff
@@ -232,397 +200,586 @@ mod tests {
+        }
+
+        #[tokio::test]
+        async fn test_temporal_types() {
+            // this test includes all supported temporal types, with the schema
+            // CREATE TABLE orders (
+            //     o_key integer,
+            //     o_time_0 time(0),
+            //     o_time_6 time(6),
+            //     o_timez_0 time(0) with time zone,
+            //     o_timez_6 time(6) with time zone,
+            //     o_timestamp_0 timestamp(0),
+            //     o_timestamp_6 timestamp(6),
+            //     o_timestampz_0 timestamp(0) with time zone,
+            //     o_timestampz_6 timestamp(6) with time zone,
+            //     o_interval interval,
+            //     o_date date,
+            //     PRIMARY KEY (o_key)
+            // );
+            // this test covers an insert event on the table above
+            let data = br#"{"payload":{"before":null,"after":{"o_key":0,"o_time_0":40271000000,"o_time_6":40271000010,"o_timez_0":"11:11:11Z","o_timez_6":"11:11:11.00001Z","o_timestamp_0":1321009871000,"o_timestamp_6":1321009871123456,"o_timestampz_0":"2011-11-11T03:11:11Z","o_timestampz_6":"2011-11-11T03:11:11.123456Z","o_interval":"P1Y2M3DT4H5M6.78S","o_date":"1999-09-09"},"source":{"version":"1.9.7.Final","connector":"postgresql","name":"RW_CDC_localhost.test.orders","ts_ms":1684733351963,"snapshot":"last","db":"test","sequence":"[null,\"26505352\"]","schema":"public","table":"orders","txId":729,"lsn":26505352,"xmin":null},"op":"r","ts_ms":1684733352110,"transaction":null}}"#;
+            let columns = get_temporal_test_columns();
+            let parser = DebeziumJsonParser::new(columns.clone(), Default::default()).unwrap();
+            let [(op, row)]: [_; 1] = parse_one(parser, columns, data.to_vec())
+                .await
+                .try_into()
+                .unwrap();
+            assert_eq!(op, Op::Insert);
+            assert!(row[0].eq(&Some(ScalarImpl::Int32(0))));
+            assert!(row[1].eq(&Some(ScalarImpl::Time(Time::new(
+                NaiveTime::from_hms_micro_opt(11, 11, 11, 0).unwrap()
+            )))));
+            assert!(row[2].eq(&Some(ScalarImpl::Time(Time::new(
+                NaiveTime::from_hms_micro_opt(11, 11, 11, 10).unwrap()
+            )))));
+            assert!(row[3].eq(&Some(ScalarImpl::Time(Time::new(
+                NaiveTime::from_hms_micro_opt(11, 11, 11, 0).unwrap()
+            )))));
+            assert!(row[4].eq(&Some(ScalarImpl::Time(Time::new(
+                NaiveTime::from_hms_micro_opt(11, 11, 11, 10).unwrap()
+            )))));
+            assert!(row[5].eq(&Some(ScalarImpl::Timestamp(Timestamp::new(
+                "2011-11-11T11:11:11".parse().unwrap()
+            )))));
+            assert!(row[6].eq(&Some(ScalarImpl::Timestamp(Timestamp::new(
+                "2011-11-11T11:11:11.123456".parse().unwrap()
+            )))));
+            assert!(
+                row[9].eq(&Some(ScalarImpl::Interval(Interval::from_month_day_usec(
```

Comment:
Why you don't use `Interval::from_iso_8601` here to construct the expect result?